These scripts were developed for ancient DNA epigenetic analysis  
Main goal: smooth methylation signal in samples and perform statistical analysis to find differentially methylated positions and regions

Scripts are designed to work with single chromosome at a time so genome needs to be splitted first.
Computing methylation profiles is to be done with DamMet (Kristian Hangh√∏j, 2019, https://gitlab.com/KHanghoj/DamMet)
Usage: bash (or sbatch) dammet_run_per_sample.sh <path-to-single-BAM-per-sample> <path-to-excluded-vcf> <output-directory-and-sample-prefix>
CpG positions file can be generated by perl script cgpositionFinder.pl and has a name like chr*.CpG.positions.txt
extract_dm.sh <path-to-F-with-name-prefix> <path-to-directory-with-CpG-reference> <out-prefix>

smoothing_dm_adding_modern.py takes methylation levels calculated by DamMet (tab delimeted txt files produced by axtract_dm.sh) and other mathylation levels (4 columns .bed where column 4 is methy;ation %) and outputs table containing positions as a first column and methylation levels of samples smoothed by calculating running mean within specified sliding window.
This script requiers two arguments: first is path to directory where all .txt and .bed files are stored and second is path to file containing CpG positions of your reference genome.
CpG positions file can be generated by perl script cgpositionFinder.pl and has a name like chr*.CpG.positions.txt



dmp_anova.py uses this table to calculate ANOVA statistics for all three groups at every position and outputs table with positions, statistics and pvalues

After that using combined p-values for DMR discovery is recommended (comb_p.sh).
https://github.com/brentp/combined-pvalues

mean_dmr.py uses region_p bed file and output table of smoothing_dm_adding_modern.py to calculate mean methylation within group in discovered regions; it prints bed format table with chromosome, region start and end as first three columns and then columns for each group of samples
